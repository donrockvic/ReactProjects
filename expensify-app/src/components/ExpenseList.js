import React from 'react';
import { connect } from 'react-redux';
import ExpenseListItem from './ExpenseListItem';
import selectExpense from '../selectors/expences';

export const ExpenseList = (props) => (
    <div>
        <h1>Expense List</h1>
        {props.expenses.length === 0} ? (
            <p> No Expense </p>
        ) : (
            {props.expenses.map((expense)=>{
                return <ExpenseListItem key={expense.id} {...expense}  />
            })}
        )
        
    </div>
);

const mapStateToProps = (state) =>{
    return {
        expenses: selectExpense(state.expenses, state.filter)
    };
};

export default  connect(mapStateToProps)(ExpenseList);


