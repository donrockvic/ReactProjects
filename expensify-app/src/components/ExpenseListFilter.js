import React from 'react';
import { connect } from 'react-redux';
import { DateRangePicker } from 'react-dates';

import {setEndDate, setStartDate, setTextFilter, sortByAmount, sortByDate} from '../actions/filters';

class ExpenseListFilters extends React.Component {
    state = {
        calFocused: null
    };

    onDatesChange = ({startDate, endDate}) => {
        this.props.dispatch(setStartDate(startDate));
        this.props.dispatch(setEndDate(endDate));
    };

    onFocusChange = (calFocused) => {
        this.setState(() => ({calFocused}))
    }

    render() {
        return (
            <div>
            <input type="text" value={this.props.filters.text} onChange={(e)=>{
                this.props.dispatch(setTextFilter(e.target.value));
            }} />
            <select value={this.props.filters.sortBy} onChange={(e) => {
                if(e.target.value === "date"){
                    this.props.dispatch(sortByDate());
                }else if(e.target.value === "amount"){
                    this.props.dispatch(sortByAmount()); 
                }
            }} >
                <option value="date">Date</option>
                <option value="amount">Amount</option>
            </select>
            <DateRangePicker
                startDate = {this.props.filters.startDate}
                endDate = {this.props.filters.endDate}
                onDatesChange={this.onDatesChange}
                focusedInput={this.state.calFocused}
                onFocusChange={this.onFocusChange}
                showClearDates={true}
                numberOfMonths={1}
                isOutsideRange={()=>false}
            />
        </div>
        )
    }
};

const mapStateToProps = (state) => {
    return {
        filters: state.filter
    };
};


export default  connect(mapStateToProps)(ExpenseListFilters);