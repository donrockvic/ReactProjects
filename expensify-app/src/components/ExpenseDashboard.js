//js
import React from 'react';

import ExpenseList from './ExpenseList';
import ExpenesListFilters from './ExpenseListFilter';

const ExpenseDashboardPage = () => (
    <div>
        <ExpenesListFilters/>
        <ExpenseList/> 
    </div>
);

export default ExpenseDashboardPage;
