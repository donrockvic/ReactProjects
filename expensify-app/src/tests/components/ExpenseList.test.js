import React from 'react';
// import shallow from 'enzyme/build/shallow';
import ReactShallowRender from 'react-test-renderer/shallow';
import {ExpenseList} from '../../components/ExpenseList';

import expences from '../fixtuers/expenses';
// import "../setupTests";

test('should render ExpenseList with expenses', () => {
    const renderer = new ReactShallowRender();
    renderer.render(<ExpenseList expenses={expences} />);
//     const wrapper = shallow ()
//     expect(wrapper).toMatchSnapshot();
expect(renderer.getRenderOutput()).toMatchSnapshot();
});


test ('should render ExpenseList with Empyty array', () => {
    const renderer = new ReactShallowRender();
    renderer.render(<ExpenseList expenses={[]} />);
expect(renderer.getRenderOutput()).toMatchSnapshot();
});
