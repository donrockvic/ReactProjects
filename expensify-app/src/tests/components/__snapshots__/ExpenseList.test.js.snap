// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseList with Empyty array 1`] = `
<div>
  <h1>
    Expense List
  </h1>
  true
   ? (
  <p>
     No Expense 
  </p>
  ) : (
  )
</div>
`;

exports[`should render ExpenseList with expenses 1`] = `
<div>
  <h1>
    Expense List
  </h1>
   ? (
  <p>
     No Expense 
  </p>
  ) : (
  <ExpenseListItem
    amount={195}
    createdAt={0}
    description="Gum"
    id="1"
    note=""
  />
  <ExpenseListItem
    amount={20000}
    createdAt={-345600000}
    description="Rent"
    id="2"
    note=""
  />
  <ExpenseListItem
    amount={4500}
    createdAt={345600000}
    description="Credit card"
    id="3"
    note=""
  />
  )
</div>
`;
